include/Zydis/DecoderTypes.h

```diff
@@ -1231,6 +1234,111 @@ typedef struct ZydisDecodedInstruction_
     } raw;
 } ZydisDecodedInstruction;
 
+/* ---------------------------------------------------------------------------------------------- */
+/* Decoder context                                                                                */
+/* ---------------------------------------------------------------------------------------------- */
+
+typedef struct ZydisDecoderContext_
+{
+    /**
+     * A pointer to the internal instruction definition.
+     */
+    const void* definition;
+    /**
+     * Contains the effective operand-size index.
+     *
+     * 0 = 16 bit, 1 = 32 bit, 2 = 64 bit
+     */
+    ZyanU8 eosz_index;
+    /**
+     * Contains the effective address-size index.
+     *
+     * 0 = 16 bit, 1 = 32 bit, 2 = 64 bit
+     */
+    ZyanU8 easz_index;
+    /**
+     * Contains some cached REX/XOP/VEX/EVEX/MVEX values to provide uniform access.
+     */
+    struct
+    {
+        ZyanU8 W;
+        ZyanU8 R;
+        ZyanU8 X;
+        ZyanU8 B;
+        ZyanU8 L;
+        ZyanU8 LL;
+        ZyanU8 R2;
+        ZyanU8 V2;
+        ZyanU8 vvvv;
+        ZyanU8 mask;
+    } vector_unified;
+    /**
+     * Information about encoded operand registers.
+     */
+    struct
+    {
+        /**
+         * Signals if the `modrm.mod == 3` or `reg` form is forced for the instruction.
+         */
+        ZyanBool is_mod_reg;
+        /**
+         * The final register id for the `reg` encoded register.
+         */
+        ZyanU8 id_reg;
+        /**
+         * The final register id for the `rm` encoded register.
+         *
+         * This value is only set, if a register is encoded in `modrm.rm`.
+         */
+        ZyanU8 id_rm;
+        /**
+         * The final register id for the `ndsndd` (`.vvvv`) encoded register.
+         */
+        ZyanU8 id_ndsndd;
+        /**
+         * The final register id for the base register.
+         *
+         * This value is only set, if a memory operand is encoded in `modrm.rm`.
+         */
+        ZyanU8 id_base;
+        /**
+         * The final register id for the index register.
+         *
+         * This value is only set, if a memory operand is encoded in `modrm.rm` and the `SIB` byte
+         * is present.
+         */
+        ZyanU8 id_index;
+    } reg_info;
+    /**
+     * Internal EVEX-specific information.
+     */
+    struct
```

Comment:
Yes, that was intentional. I thought about this for quite some time. As we rely on the user to allocate our structs, we have to make sure the input struct size matches our internal expectations. In a pure C/C++ environment, we could probably `ifdef`-ing these members away without any problem, but that's a different story for bindings. We would always have to check if the struct size is matching (e.g. going the Windows API way and introducing a `size` field in the `ZydisDecodedInstruction` that needs to be initialized by the user - extra complexity which I don't really like).

I think we might want to discuss this in detail. One good solution might be to introduce a `ZYDIS_STRICT_C_UNSAFE_FOR_BINDINGS` option. This would indicate that `Zydis` is used in a pure C/C++ project and always built on demand (-> we know the feature defines of headers and code files are the same). This would as well allow us to e.g. change the `ZydisDecodedOperand` struct to `unions` or to introduce bit-fields in the public API.

Reply:
> I think we should discuss this in detail. 

Agreed, but this definitely deserves a dedicated issue for proper discussion -- let's not do that in this PR.

Reply:
> let's not do that in this PR

Definitely not! Will create a separate issue tomorrow.